### PR TITLE
Change default values to give best experience OOTB

### PR DESCRIPTION
**Pull request by vboulineau**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has beed updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/values.yaml [hunk 1]
```diff
@@ -75,7 +75,7 @@ datadog:
   ## Autodiscovery via Kube Service annotations is automatically enabled
   clusterChecks:
     # datadog.clusterChecks.enabled -- Enable the Cluster Checks feature on both the cluster-agents and the daemonset
-    enabled: false
```
Comment (vboulineau) [inline charts/datadog/values.yaml:78]: Should we activate CLC by default?

File: charts/datadog/values.yaml [hunk 2]
```diff
@@ -141,14 +141,14 @@ datadog:
 
     # datadog.dogstatsd.nonLocalTraffic -- Enable this to make each node accept non-local statsd traffic
     ## ref: https://github.com/DataDog/docker-dd-agent#environment-variables
-    nonLocalTraffic: false
+    nonLocalTraffic: true
```
Comment (clamoriniere) [inline charts/datadog/values.yaml:144]: this is set to false on purpose for security reason: mainly because if a node is not properly protected by a firewall, anyone can push traces to the agent.

File: charts/datadog/values.yaml [hunk 3]
```diff
@@ -377,7 +376,7 @@ clusterAgent:
 
     # clusterAgent.metricsProvider.useDatadogMetrics -- Enable usage of DatadogMetric CRD to autoscale on arbitrary Datadog queries
     ## NOTE: You need to install the `DatadogMetric` CRD before
-    useDatadogMetrics: false
+    useDatadogMetrics: true
```
Comment (clamoriniere) [inline charts/datadog/values.yaml:379]: I think it is ok, but we need to document that we only supported kubernetes > 1.10 (to support crds)

File: charts/datadog/values.yaml (hunk 2 above)
Comment (vboulineau) [reply charts/datadog/values.yaml:144]: Not sure as `useHostPort: false` by default

File: charts/datadog/values.yaml (hunk 3 above)
Comment (vboulineau) [reply charts/datadog/values.yaml:379]: So we agree to drop Kube < 1.10 (I think ATM we support 1.8?)

File: charts/datadog/values.yaml (hunk 1 above)
Comment (clamoriniere) [reply charts/datadog/values.yaml:78]: no. we will activate only the cluster-agent by default

**Comment by clamoriniere** [comment]
@vboulineau 
any reason why this PR is still a draft pr?

**Comment by vboulineau** [comment]
> @vboulineau
> any reason why this PR is still a draft pr?

Not anymore :)

**Review by clamoriniere — COMMENTED** [review]
/LGTM

could you also update the default agent tag?

File: charts/datadog/templates/_container-agent.yaml [hunk 4]
```diff
@@ -45,7 +45,7 @@
       value: {{ tpl (.Values.datadog.dogstatsd.tags | join " " | quote) . }}
     {{- end }}
     {{- if not .Values.clusterAgent.enabled }}
-    {{- if .Values.datadog.leaderElection }}
+    {{- if and (not .Values.clusterAgent.enabled) .Values.datadog.leaderElection }}
```
Comment (hkaj) [inline charts/datadog/templates/_container-agent.yaml:48]: is this change needed? The line directly above already ensures the `not .Values.clusterAgent.enabled` condition

Comment (vboulineau) [reply charts/datadog/templates/_container-agent.yaml:48]: Indeed, good catch, I must have missed it! Fixed

**Review by xornivore — APPROVED** [review]

File: charts/datadog/templates/_helpers.tpl [hunk 5]
```diff
@@ -197,3 +197,14 @@ true
 false
 {{- end -}}
 {{- end -}}
+
+{{/*
+Return true if the system-probe container should be created.
```
Comment (hkaj) [inline charts/datadog/templates/_helpers.tpl:202]: copy pasta? :D

Comment (vboulineau) [reply charts/datadog/templates/_helpers.tpl]: Credo di sì :D
